lib/board.rb

```diff
@@ -46,12 +46,20 @@ def win?
             end
         end
     end
-end
-# matrix = []
-        # Matrix.rows(@board).each do |row|
-        #     # matrix << diagonal
-        #     require 'pry';binding.pry
-        # end
-        (0..3) (i=var.length; i--) do
-            (0..4) (j=another_var.length; j--)
-        end
\ No newline at end of file
+
+    def four_in_a_row_by_row(board)
+        board.each do |row|
+            a = row.each_cons(4).find { |a| a.uniq.size == 1 && a.first != '.' }
+            return true unless a.nil?        
+        end
+        nil
+    end
+
+    def diagonals(board)
```

Comment:
This is perfect. Now we can use just this method to check diagonally across the board, then rotate the board to check the opposite diagonal direction without having to write a separate method.